### PR TITLE
forbid invalid edge weights and negative default weight in K_SHORTEST…

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This is a follow up PR on: https://github.com/arangodb/arangodb/pull/14178 (A negative edge weight can crash the server in a weighted traversal.)

As this referenced issue also persists in `K_SHORTEST_PATHS` and `SHORTEST_PATH`, this PR adds the same control mechanism to them to make sure no negative edge weights are going to be used (invalid), whether set in an edge directly or set via default weight attribute. 

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] Backports required for: 3.8 

#### Related Information

- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-409

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)

Link to Jenkins PR run:

### Documentation

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API

### CODE REVIEW

File: arangod/Graph/ShortestPathOptions.cpp [hunk 1]
```diff
@@ -162,8 +160,13 @@ void ShortestPathOptions::addReverseLookupInfo(aql::ExecutionPlan* plan,
 
 double ShortestPathOptions::weightEdge(VPackSlice edge) const {
   TRI_ASSERT(useWeight());
-  return arangodb::basics::VelocyPackHelper::getNumericValue<double>(
-      edge, weightAttribute.c_str(), defaultWeight);
+  const auto weight = arangodb::basics::VelocyPackHelper::getNumericValue<double>(
+      edge, _weightAttribute.c_str(), _defaultWeight);
+  if (weight < 0.) {
```
Comment (maierlars) [inline arangod/Graph/ShortestPathOptions.cpp:165]:
```suggestion
  if (weight <= 0.) {
```

File: arangod/Graph/ShortestPathOptions.cpp [hunk 2]
```diff
@@ -212,18 +215,41 @@ auto ShortestPathOptions::estimateDepth() const noexcept -> uint64_t {
   return 7;
 }
 
+auto ShortestPathOptions::setDefaultWeight(double weight) -> void {
+  if (weight < 0.) {
```
Comment (maierlars) [inline arangod/Graph/ShortestPathOptions.cpp:219]:
```suggestion
  if (weight <= 0.) {
```

File: arangod/Graph/ShortestPathOptions.cpp [hunk 3]
```diff
@@ -212,18 +215,41 @@ auto ShortestPathOptions::estimateDepth() const noexcept -> uint64_t {
...
       end{other.end},
-      weightAttribute{other.weightAttribute},
-      defaultWeight{other.defaultWeight},
       bidirectional{other.bidirectional},
       multiThreaded{other.multiThreaded},
-      _reverseLookupInfos{other._reverseLookupInfos} {}
+      _reverseLookupInfos{other._reverseLookupInfos},
+      _weightAttribute{other._weightAttribute},
+      _defaultWeight{other._defaultWeight} {
+  TRI_ASSERT(other._defaultWeight >= 0.);
```
Comment (maierlars) [inline arangod/Graph/ShortestPathOptions.cpp:251]:
```suggestion
  TRI_ASSERT(other._defaultWeight > 0.);
```

**Review by maierlars — APPROVED** [review]

File: arangod/Graph/ShortestPathOptions.cpp (hunk 3 above)
Comment (hkernbach) [reply arangod/Graph/ShortestPathOptions.cpp:251]: TY. will apply in separate commit as I am lazy and want to cherry-pick in a single commit :)

File: arangod/Graph/ShortestPathOptions.cpp (hunk 2 above)
Comment (hkernbach) [reply arangod/Graph/ShortestPathOptions.cpp:219]: TY. will apply in separate commit as I am lazy and want to cherry-pick in a single commit :)

File: arangod/Graph/ShortestPathOptions.cpp (hunk 1 above)
Comment (hkernbach) [reply arangod/Graph/ShortestPathOptions.cpp:165]: TY. will apply in separate commit as I am lazy and want to cherry-pick in a single commit :)

File: arangod/Graph/ShortestPathOptions.cpp (hunk 3 above)
Comment (hkernbach) [reply arangod/Graph/ShortestPathOptions.cpp]: reverted

File: arangod/Graph/ShortestPathOptions.cpp (hunk 2 above)
Comment (hkernbach) [reply arangod/Graph/ShortestPathOptions.cpp]: reverted

**Pull request merged**